.github/workflows/ci.yml

```diff
@@ -331,14 +331,136 @@ jobs:
+      docker: ${{ steps.determine.outputs.docker }}
+      nesqueit: ${{ steps.determine.outputs.nesqueit }}
+      helm: ${{ steps.determine.outputs.helm }}
+    steps:
+      - name: install minimatch
+        shell: bash
+        run: npm install minimatch@"^7.4.3"
+      - name: Determine conditional jobs to run
+        id: determine
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const {Minimatch} = require("minimatch");
+
+            async function getChangedFiles(pull_request) {
+              const listFilesOptions = github.rest.pulls.listFiles.endpoint.merge({
+                owner: pull_request.base.repo.owner.login,
+                repo: pull_request.base.repo.name,
+                pull_number: pull_request.number,
+              });
+              return github.paginate(listFilesOptions).then(resp => resp.map(f => f.filename))
+            }
+
+            function isMatch(changedFile, matchers) {
+              for (const matcher of matchers) {
+                if (matcher.match(changedFile)) {
+                  core.info("Match found for changed file " + changedFile);
+                  return true;
+                }
+              }
+              core.info("No changed file matched the globs");
+              return false;
+            }
+
+            function filesMatchAnyGlob(changedFiles, globs) {
+              const matchers = globs.map(g => new Minimatch(g));
+              for (const changedFile of changedFiles) {
+                if (isMatch(changedFile, matchers)) {
+                  return true;
+                }
+              }
+              return false;
+            }
+
+            function checkLabelOrGlobs(pr, changedFiles, labelName, globs) {
+              core.info("Checking for " + labelName);
+
+              // If the PR has the given label, then let the conditional job run.
+              for (const label of pr.labels) {
+                if (label.name === labelName) {
+                  core.info("PR has label " + labelName);
+                  return true;
+                }
+              }
+
+              // For pull request, check the list of changed files against the given globs.
+              // Let the conditional job run, if any of the globs matches and of the changed files.
+              core.info("Checking changed files against globs " + globs);
+              return filesMatchAnyGlob(changedFiles, globs);
+            }
+
+            ///////////////////////////////////////////////////////////////////////////////////////
+            // "Non PR" case:
+            // Behavior of the conditional jobs when the event that triggered the workflow is not
+            // for a pull request (usually a push-to-main).
+            //
+            if (!context.payload || !context.payload.pull_request) {
+              core.info("Not a pull-request, enabling all jobs.");
+
+              // Yield 'true' for all event payload types, except pull_request
```

Comment:
but `nesqueit` is `false`?

Reply:
yes, handled in a scheduled job - updated comments